js/libs/ui-shared/src/user-profile/MultiInputComponent.tsx

```diff
@@ -0,0 +1,150 @@
+import {
+  Button,
+  ButtonVariant,
+  InputGroup,
+  TextInput,
+  TextInputProps,
+} from "@patternfly/react-core";
+import { MinusCircleIcon, PlusCircleIcon } from "@patternfly/react-icons";
+import { Fragment, useEffect, useMemo } from "react";
+import { FieldPath, UseFormReturn, useWatch } from "react-hook-form";
+import { UserProfileFieldProps } from "./UserProfileFields";
+import { UserProfileGroup } from "./UserProfileGroup";
+import { TranslationFunction, UserFormFields, fieldName, label } from "./utils";
+
+export const MultiInputComponent = ({
+  t,
+  form,
+  attribute,
+  renderer,
+}: UserProfileFieldProps) => (
+  <UserProfileGroup t={t} form={form} attribute={attribute} renderer={renderer}>
+    <MultiLineInput
+      t={t}
+      form={form}
+      aria-label={label(attribute, t)}
+      name={fieldName(attribute.name)!}
+      addButtonLabel={t("addMultivaluedLabel", {
+        fieldLabel: label(attribute, t),
+      })}
+    />
+  </UserProfileGroup>
+);
+
+function stringToMultiline(value?: string): string[] {
+  return typeof value === "string" ? value.split("##") : [];
+}
+
+function toStringValue(formValue: string[]): string {
+  return formValue.join("##");
+}
+
+export type MultiLineInputProps = Omit<TextInputProps, "form"> & {
+  t: TranslationFunction;
+  name: FieldPath<UserFormFields>;
+  form: UseFormReturn<UserFormFields>;
+  addButtonLabel?: string;
+  isDisabled?: boolean;
+  defaultValue?: string[];
+  stringify?: boolean;
+};
+
+const MultiLineInput = ({
+  t,
+  name,
+  form,
+  addButtonLabel,
+  isDisabled = false,
+  defaultValue,
+  stringify = false,
+  id,
+  ...rest
+}: MultiLineInputProps) => {
+  const { register, setValue, control } = form;
+  const value = useWatch({
+    name,
+    control,
+    defaultValue: defaultValue || "",
+  });
+
+  const fields = useMemo<string[]>(() => {
```

Comment:
Same here, I do believe we no longer handle multi-value with string delimiters when user profile is enabled.